xbmc/FileItemList.cpp

```diff
@@ -718,279 +742,164 @@ void CFileItemList::Stack(bool stackFiles /* = true */)
   // items needs to be sorted for stuff below to work properly
   Sort(SortByLabel, SortOrderAscending);
 
-  StackFolders();
-
-  if (stackFiles)
-    StackFiles();
-}
+  // Convert folder paths containing disc images to files (INDEX.BDMV or VIDEO_TS.IFO)
+  ConvertDiscFoldersToFiles();
 
-void CFileItemList::StackFolders()
-{
-  // Precompile our REs
+  // Get REs
   VECCREGEXP folderRegExps;
-  CRegExp folderRegExp(true, CRegExp::autoUtf8);
-  const std::vector<std::string>& strFolderRegExps =
-      CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_folderStackRegExps;
-
-  std::vector<std::string>::const_iterator strExpression = strFolderRegExps.begin();
-  while (strExpression != strFolderRegExps.end())
-  {
-    if (!folderRegExp.RegComp(*strExpression))
-      CLog::Log(LOGERROR, "{}: Invalid folder stack RegExp:'{}'", __FUNCTION__,
-                strExpression->c_str());
-    else
-      folderRegExps.push_back(folderRegExp);
-
-    ++strExpression;
-  }
+  VECCREGEXP fileRegExps;
+  CStackDirectory::LoadRegExps(fileRegExps, folderRegExps);
 
-  if (!folderRegExp.IsCompiled())
+  if (fileRegExps.empty() || folderRegExps.empty())
+    CLog::LogF(LOGDEBUG, "No stack expressions available. Skipping stacking");
+  else
   {
-    CLog::Log(LOGDEBUG, "{}: No stack expressions available. Skipping folder stacking",
-              __FUNCTION__);
-    return;
-  }
+    // Find stack candidates (ie. any file or directory that matches a RegEx)
+    std::vector<std::tuple<int, std::string, std::string, int, int64_t>> stackCandidates;
```

Comment:
A comment explaining what exactly the tuple contains would really help the code reader.